File: 151003/Alekseeva/TaskREST/TaskREST/Mappers/TweetMapper.cs
using Riok.Mapperly.Abstractions;
using TaskREST.Dto.Request.CreateTo;
using TaskREST.Dto.Request.UpdateTo;
using TaskREST.Dto.Response;
using TaskREST.Models;

namespace TaskREST.Mappers;

[Mapper]
public static partial class TweetMapper
{
    public static partial Tweet Map(UpdateTweetRequestTo updateTweetRequestTo);
    public static partial Tweet Map(CreateTweetRequestTo createTweetRequestTo);
    public static partial TweetResponseTo Map(Tweet tweet);
    public static partial IEnumerable<TweetResponseTo> Map(IEnumerable<Tweet> tweets);

    public static partial IEnumerable<Tweet> Map(
        IEnumerable<UpdateTweetRequestTo> tweetRequestTos);
}

File: 151002/Bybikov/SecondJPASolution/src/main/java/com/example/rv/impl/tweet/mapper/TweetMapper.java
package com.example.rv.impl.tweet.mapper;

import com.example.rv.impl.editor.Editor;
import com.example.rv.impl.tweet.Tweet;
import com.example.rv.impl.tweet.dto.TweetRequestTo;
import com.example.rv.impl.tweet.dto.TweetResponseTo;

import java.util.List;

public interface TweetMapper {

    TweetRequestTo tweetToRequestTo(Tweet tweet);

    List<TweetRequestTo> tweetToRequestTo(Iterable<Tweet> tweets);

    Tweet dtoToEntity(TweetRequestTo tweetRequestTo, Editor editor);

    TweetResponseTo tweetToResponseTo(Tweet tweet);

    List<TweetResponseTo> tweetToResponseTo(Iterable<Tweet> tweets);
}
